data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/consumer/OrderedConsumerVerticle.java

```diff
@@ -223,4 +245,42 @@ private boolean areAllExecutorsBusy() {
     // All executors are busy
     return true;
   }
+
+  private void recordExecutorQueueLatency(final ConsumerRecordContext recordContext) {
+    final var latency = recordContext.performLatency();
+    logger.debug("Executor queue latency {}", keyValue("latency", latency));
+
+    if (meterRegistry != null) {
+      Metrics
+      .executorQueueLatency(getTags(recordContext))
+      .register(meterRegistry)
+      .record(latency);
+    }
+  }
+
+  private void setValueQueueLength(final TopicPartition topicPartition, Queue<Supplier<Future<?>>> executor) {
+    if (meterRegistry != null) {
+      Metrics
+        .queueLength(getTags(topicPartition), executor)
+        .register(meterRegistry)
+        .value();
+    }
+  }
+
+  private Tags getTags(ConsumerRecordContext recordContext) {
+    if (recordContext.getRecord().record().value() instanceof InvalidCloudEvent) {
+      return Tags.of(
+        Tag.of(Metrics.Tags.EVENT_TYPE, "InvalidCloudEvent")
+      );
+    }
+    return Tags.of(
+      Tag.of(Metrics.Tags.EVENT_TYPE, recordContext.getRecord().value().getType())
+    );
+  }
```

Comment:
to undestand to which Kafka API resource it belongs, we need to add tags with the resource name/namespace, we can get them using `this.egress.getReference().getName()` and  `this.egress.getReference().getNamespace()`

```java
      Tag.of(Metrics.Tags.RESOURCE_NAMESPACE, egress.getReference().getNamespace()),
      Tag.of(Metrics.Tags.CONSUMER_NAME, egress.getReference().getName())
```

Reply:
great - thank you! Didn't notice that `egress` was a field in this class already.